core/src/main/java/tech/tablesaw/api/Row.java

```diff
@@ -82,64 +84,110 @@ public boolean hasNext() {
         return table.columnNames();
     }
 
-    public int columnCount() {
-        return table.columnCount();
+    public Boolean getBoolean(int columnIndex) {
+        return getBoolean(columnNames[columnIndex]);
     }
 
-    @Override
-    public Row next() {
-        rowNumber++;
-        return this;
+    public Boolean getBoolean(String columnName) {
+        return booleanColumnMap.get(columnName).get(rowNumber);
     }
 
-    public double getDouble(String columnName) {
-        return doubleColumnMap.get(columnName).getDouble(rowNumber);
+    public LocalDate getDate(String columnName) {
+        return dateColumnMap.get(columnName).get(rowNumber);
+    }
+
+    public LocalDate getDate(int columnIndex) {
+        return dateColumnMap.get(columnNames[columnIndex]).get(rowNumber);
+    }
+
+    public LocalDateTime getDateTime(int columnIndex) {
+        return getDateTime(columnNames[columnIndex]);
+    }
+
+    public LocalDateTime getDateTime(String columnName) {
+        return ((DateTimeColumn) columnMap.get(columnName)).get(rowNumber);
     }
 
     public double getDouble(int columnIndex) {
         return getDouble(columnNames[columnIndex]);
     }
 
-    public int getInt(String columnName) {
-    	return intColumnMap.get(columnName).getInt(rowNumber);
+    public double getDouble(String columnName) {
+        return doubleColumnMap.get(columnName).getDouble(rowNumber);
+    }
+
+    public float getFloat(int columnIndex) {
+        return getFloat(columnNames[columnIndex]);
+    }
+
+    public float getFloat(String columnName) {
+        return floatColumnMap.get(columnName).getFloat(rowNumber);
     }
 
     public int getInt(int columnIndex) {
         return getInt(columnNames[columnIndex]);
     }
 
-    public long getLong(String columnName) {
-	return longColumnMap.get(columnName).getLong(rowNumber);
+    public int getInt(String columnName) {
+        return intColumnMap.get(columnName).getInt(rowNumber);
     }
 
     public long getLong(int columnIndex) {
         return getLong(columnNames[columnIndex]);
     }
 
-    public short getShort(String columnName) {
-	return shortColumnMap.get(columnName).getShort(rowNumber);
+    public long getLong(String columnName) {
+        return longColumnMap.get(columnName).getLong(rowNumber);
     }
 
-    public short getShort(int columnIndex) {
-        return getShort(columnNames[columnIndex]);
+    public Object getObject(String columnName) {
+        return columnMap.get(columnName).get(rowNumber);
     }
 
-    public float getFloat(String columnName) {
-	return floatColumnMap.get(columnName).getFloat(rowNumber);
+    public Object getObject(int columnIndex) {
+        return columnMap.get(columnNames[columnIndex]).get(rowNumber);
     }
 
-    public float getFloat(int columnIndex) {
-        return getFloat(columnNames[columnIndex]);
+    public int getPackedDate(String columnName) {
```

Comment:
I would probably not include getters for the packed date / datetime as I think it's a bit of an implementation detail

Reply:
I think i've used them for performance sensitive work, but i agree they'd almost never be needed in normal use. i'm ok with their being removed.

Reply:
I'll add an issue for this and do it in a subsequent change